lib/decanter/core.rb

```diff
@@ -129,20 +130,24 @@ def unhandled_keys(args)
       def handled_keys(args)
         handlers.reduce({}) do |acc, curr|
           name, handler = *curr
+
           values = args.values_at(*name)
-          values = values.length == 1 ? values.first : values
+          values = values.length == 1 ? values.first : values                    
 
-          is_empty_input = Array(values).all?(&:blank?)
-          if is_empty_input
+          if key_is_missing?(handler, args)
             empty_required_input_error(name) if handler[:options][:required]
-            # Skip handling empty inputs
-            next acc
+            next acc # Skip handling empty inputs
           end
 
           acc.merge handle(handler, values)
         end
       end
 
+      def key_is_missing?(handler, args)
+        in_keys = Array(handler[:in_key]) # squash parser has multiple in_key keys
```

Comment:
Can you use `name` here as defined above rather than adding `in_key` to the handler hash?